docs/src/main/asciidoc/oauth2-guide.adoc

```diff
@@ -0,0 +1,357 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
+////
+= Quarkus - Using OAuth2 RBAC
+
+include::./attributes.adoc[]
+:extension-name: Elytron Security OAuth2
+:oauth2-token: OAuth2 tokens
+
+This guide explains how your Quarkus application can utilize {oauth2-token} to provide secured access to the JAX-RS endpoints.
+
+If your OAuth2 Authentication server provides JWT tokens, you should use link:jwt-guide.html[MicroProfile JWT RBAC] instead, this extension aims to be used with obscure tokens and validate the token by calling an introspection endpoint.
```

Comment:
`obscure`?

Reply:
`obscure` tokens are the opposite of meaningfull tokens like JWT ones as they didn't carry any informations. This terms comes from the discussion on this PR, maybe I can replace it as it's not a commonly used one but I don't know what to use ?

Reply:
`exotic` maybe if it's the meaning you want to convey?

Reply:
Isn't the correct term `opaque token` ?

Reply:
@sebastienblanc yes, `opaque` is the correct term, I will update the documentation with this.

Reply:
done